src/niamoto/core/components/imports/plots.py

```
@@ -314,3 +357,368 @@
+                            progress.update(task, advance=1)
+
+            return total_linked
+
+        except Exception as e:
+            from sqlalchemy.exc import SQLAlchemyError
+
+            if isinstance(e, SQLAlchemyError):
+                raise DatabaseError(
+                    "Database error during occurrence linking",
+                    details={"error": str(e)},
+                )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)